assignment-2/src/presenter/sOUserPresenter.js

```diff
@@ -0,0 +1,87 @@
+import sOUserModel from "../model/sOUserModel";
+
+class SOUserPresenter {
+    onLogin = () => {
+        let newSOUser = sOUserModel.state.newSOUser;
+        let sOUsers = sOUserModel.state.sOUsers;
+        let loginSuccess = false;
+
+        for (let i = 0; i < sOUsers.length; i++) {
```

Comment:
This should be part of the model (exposed via a method).

Reply:
My bad, I thought that the presenter should have had the logic to change the states in model. I will move it inside the model.